### DISCUSSION TITLE
How to stop internal metrics from OTel Collector into stdout/console output

### DISCUSSION THREAD

**Discussion by sahangunathilaka** [open-telemetry/opentelemetry-collector-contrib#36792] [Q&A]
I have a OTel Collector deployment in Azure Container App service. I'm using otel/opentelemetry-collector-contrib:latest image.
Can someone share me how to disable to internal metrics logged into stdout/console output from OTel Collector itself please. I have tried setting the metrics log level to with the following config. But it doesn't work.
```
service:
  telemetry:
      metrics:
        level: none
```

Following is a sample metrics information I'm getting into my console logs. I need to disable them.
```
StartTimestamp: 2024-12-11 01:22:19.167 +0000 UTC
Timestamp: 2024-12-11 12:50:19.167 +0000 UTC
Value: 0.000000
Metric #14
Descriptor:
     -> Name: otelcol_process_memory_rss
     -> Description: Total physical memory (resident set size) [alpha]
     -> Unit: 
     -> DataType: Gauge
NumberDataPoints #0
Data point attributes:
     -> service_instance_id: Str(cc09e3fa-0301-4ccb-9045-41e81ed40701)
     -> service_name: Str(otelcol-contrib)
     -> service_version: Str(0.115.0)
```

Below is my simple config.yaml.
```
exporters:
  debug:
    verbosity: detailed

receivers:
  otlp:
    protocols:
      grpc:
      http:

processors:
  attributes/enrich-environment:
    actions:
      - key: collector.test
        value: test
        action: insert

service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: [attributes/enrich-environment]
      exporters: [debug]
    metrics:
      receivers: [otlp]
      processors: [attributes/enrich-environment]
      exporters: [debug]
    logs:
      receivers: [otlp]
      processors: [attributes/enrich-environment]
      exporters: [debug]
  telemetry:
    metrics:
      level: none
```

Any input is highly appreciated. Thanks!

**Comment by gabrielSanchezCampello** [comment]
Good morning, 

I think it is due to the debug exporter with verbosity detailed.
https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/debugexporter/README.md

If you set verbosity normal or basic you should stop seeing so many logs, and if you remove that exporter you will stop seeing everything straight away.

I hope that helps.

**Reply by sahangunathilaka** [reply]
Hi @gabrielSanchezCampello
Thank you for looking into it. But I can confirm it is not from the debug exporter. I've already tried removing debug exporter and still can see internal telemetry in console logs. Also, I do need to enable debug exporter validate my OTel Collector functionality. So, I want to enable debug exporter and just disable internal telemetry as OTel documentation mentions. Appreciate your time again.

**Reply by nivethsaran** [reply]
Hi @sahangunathilaka,
I know it has been a while, but I am facing the exact same issue, were you able to find a solution to this problem?

**Reply by sahangunathilaka** [reply]
> Hi @sahangunathilaka, I know it has been a while, but I am facing the exact same issue, were you able to find a solution to this problem?

Hi @nivethsaran,
I tested this out with a docker image available in the Github registry. The same configuration worked for that image. I cannot recall the exact version, but somewhere around 0.120.0 - 0.130.0 tag version of the image - https://github.com/open-telemetry/opentelemetry-collector-releases/pkgs/container/opentelemetry-collector-releases%2Fopentelemetry-collector-contrib worked fine.
Try using one of the images from the above link and using the following configuration block if you still encounter the issue.
```
service:
  telemetry:
      metrics:
        level: none
```
